Refactored Results Table to Use Semantic-UI

cormacjmollitor opened the pull request:
Filter sidebar needs to be refactored as well. Using the React Semantic-UI component, it goes through the navbar so once the Semantic-UI changes to the navbar are integrated, we can start working on that

Closes #172

cormacjmollitor committed:
Refactored Results Table to Use Semantic-UI

Filter sidebar needs to be refactored as well. Using the React Semantic-UI component, it goes through the navbar so once the Semantic-UI changes to the navbar are integrated, we can start working on that

Closes #172

frostyshadows commented on client/src/components/ResultsTable.js:
```diff
@@ -1,15 +1,85 @@
+  }
+
+  // Takes 2 api calls to fire
+  componentWillReceiveProps() {
```
Weird, is this supposed to take 2 api calls to fire or just something you had to do to make it work? If it's the latter we should try to figure out why. It might be something in the addActiveFields() implementation or using the wrong React component lifecycle callback.

frostyshadows commented on client/src/components/ResultsTable.js:
```diff
@@ -1,15 +1,85 @@
+                  </Accordion>
+                </Item.Description>
+                <Item.Extra>
+                  <Label icon='globe' content={result.city + ", " + result.country} />
```
you should be able to use a string template here

frostyshadows reviewed (changes requested):
Looked good when I ran it! Left a few comments. Could you add some padding to the top? And the corners in the designs are slightly more rounded.
<img width="1440" alt="screen shot 2019-02-08 at 8 53 11 pm" src="https://user-images.githubusercontent.com/12488887/52516643-96dac100-2be3-11e9-8738-50dcee21d3be.png">

cormacjmollitor committed:
Added Padding to Results Table

cormacjmollitor committed:
Active Field Not Updating

cormacjmollitor committed:
Requested Changes to Pull Request #184

Moved the code for individual results from ResultsTable.js to ResultRow.js and deleted ResultData.js. Some of the problems in the code from the prior commits on this branch were solved by giving each result their own component and thus their own state.

cormacjmollitor commented:
@frostyshadows everything should be good now if you wanna take a look

frostyshadows commented on client/src/components/ResultRow.js:
```diff
@@ -7,44 +14,44 @@ class ResultRow extends Component {
-    }
-  };
+
+  setSelected = () => {
```
could rename this to toggleSelected() since it's more of a toggle than a setter

frostyshadows reviewed (approved):
LGTM!
